src/openvtb/components/verzoeken/api/validators.py

```diff
@@ -85,3 +88,47 @@ def __call__(self, new_value, serializer_field):
 
         if new_value != current_value:
             raise serializers.ValidationError(self.message, code=self.code)
+
+
+class AanvraagGegevensValidator:
+    """
+    Validates `aanvraag_gegevens` against the JSON Schema for the selected
+    `VerzoekType` and `Version`.
+
+    Raises a `ValidationError` if the schema version is unknown or if
+    JSON Schema validation fails.
+    """
+
+    code = "invalid-json-schema"
+    label = "aanvraagGegevens"
+    requires_context = True
+
+    def __call__(self, attrs, serializer):
+        verzoek_type = get_from_serializer_data_or_instance(
+            "verzoek_type", attrs, serializer
+        )
+        aanvraag_gegevens = get_from_serializer_data_or_instance(
+            "aanvraag_gegevens", attrs, serializer
+        )
+        version = get_from_serializer_data_or_instance("version", attrs, serializer)
+        if version not in verzoek_type.versions.values_list("version", flat=True):
```

Comment:
I think using `.exists` is slightly more efficient here, because in that case the database stops executing the query as soon as it finds a matching row, and the resulting data that is transferred is just a boolean
```suggestion
        if not verzoek_type.versions.filter(version=version).exists()
```